allow_price_update_on_locked_so/models/sale_order_line.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import AccessError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _get_protected_fields(self):
+        """
+        Override to remove 'price_unit' from the list of protected fields
+        for users with specific permissions.
+        """
```

Comment:
This method has a docstring already
we just use the `comment` here to understand the purpose of override